### ISSUE TITLE
Jetpack Focus: Incorrect app state after logging out of WP.com account with a self-hosted site on WP app

### ISSUE THREAD

**Issue by staskus** [wordpress-mobile/WordPress-Android#19097]
### Expected behavior

After logging out of the WP.com account with a self-hosted site on WP app, we should show the self-hosted site menu and no tabs

### Actual behavior

WP app shows an interface with tabs and indicates that there's no site until closing and reopening the app

### Steps to reproduce the behavior

1. Open WordPress app
2. Log into self-hosted site
3. Log into WP.com account
4. Logout of WP.com account

##### Tested on Pixel 4, Android [version], WPAndroid 23.1-rc2

https://github.com/wordpress-mobile/WordPress-Android/assets/4062343/67fab93c-99db-4eb6-97f8-9e1a4158ad09

**Comment by jarvislin** [comment]
I tried to reproduce the issue, but I got a different result.

1. Open WordPress app
2. Log into self-hosted site
3. Log into WP.com account
4. Logout of WP.com account
5. It redirects me to the login screen. <----

Redirect users to the login screen looks make sense, too. I'm not sure if there have been any recent adjustments to the behavior after logging out of WordPress.com.

Tested on WordPress Beta 23.9, Pixel 3a
